src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,88 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable no-lone-blocks */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  selectedTodoId: number,
+  closeModal: () => void,
+};
+
+export const TodoModal: React.FC<Props> = (
+  {
+    todos,
+    selectedTodoId,
+    closeModal,
+  },
+) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    const foundTodo = todos.find(todo => todo.id === selectedTodoId);
+
+    if (foundTodo) {
+      setSelectedTodo(foundTodo);
+    }
+  }, [selectedTodo]);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId).then(setUser);
+    }
+  }, [selectedTodo]);
+  {
+    if (!user) {
+      return <Loader />;
```

Comment:
it is more correct to immediately show the popup, and in it the loader, while the data is requested from the server